### PR TITLE
test: Disable dshell/dll.d test on OSX

### PR DESCRIPTION
**Pull request by ibuclaw**
When running the dll test, the [following](https://github.com/dlang/druntime/blob/master/src/rt/sections_osx_x86_64.d#L139) warning is written to console.

```
user@darwin20:~/src/dlang/dmd/test$ ./test_results/dshell/dll/testdll
Shared libraries are not yet supported on OSX.
```

This is correct, as when you link druntime into a shared library on OSX, all the symbols of libphobos.a are copied into the DSO/image for that library.  Then, linking an executable with both libphobos and that library means there are now two copies of all globals present at run-time.

i.e:
```
user@darwin20:~/src/dlang/dmd/test$ otool -L test_results/dshell/dll/testdll
test_results/dshell/dll/testdll:
	test_results/dshell/dll/mydll.so (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.100.5)

user@darwin20:~/src/dlang/dmd/test$ nm test_results/dshell/dll/testdll | grep __D4core6thread10threadbase10ThreadBase6_slockG96v
00000001000399e8 B __D4core6thread10threadbase10ThreadBase6_slockG96v

user@darwin20:~/src/dlang/dmd/test$ nm test_results/dshell/dll/mydll.so | grep __D4core6thread10threadbase10ThreadBase6_slockG96v
0000000000057c58 B __D4core6thread10threadbase10ThreadBase6_slockG96v
```

Unlike ELF DSOs, where at run-time these are resolved as-if the symbol is weak.  On OSX, depending on which DSO/image a referencing function is resolved/called from, you could get either version of these symbols. i.e: `Thread.initLocks()` is called from `testdll.exe`, initializing the `Thread.slock@1000399e8` handle in that image, but `thread_suspendAll` is called from `mydll.so`, which tries to use the uninitialized `Thread.slock@000057c58` and throws an Error as a result.

Until DMD sorts out symbol emission to avoid this (GDC annotates these `__gshared` variables with `@weak` IIRC), or gains proper support for shared libraries, it can only be considered a fluke that this test even passes.  Disabling it unblocks #14194.

### CODE REVIEW

File: test/dshell/dll.d [hunk 1]
```diff
@@ -5,6 +5,12 @@ int main()
     version (Windows) if (Vars.MODEL == "32omf") // Avoid optlink
         return DISABLED;
 
+    version (OSX) // Shared libraries are not yet supported on OSX
```
Comment (thewilsonator) [inline test/dshell/dll.d:8]: If this is a DMD only problem, this should probably be `version(DigitalMars) version (OSX)` rather that just `version(OSX)`

**Comment by WalterBright** [comment]
I added a bugzilla for it: https://issues.dlang.org/show_bug.cgi?id=23180

File: test/dshell/dll.d (hunk 1 above)
Comment (WalterBright) [reply test/dshell/dll.d:8]: Since GDC resolves it with the @weak attribute, I agree with @thewilsonator

Comment (ibuclaw) [reply test/dshell/dll.d:8]: I don't run the dshell tests, does ldc?

Comment (thewilsonator) [reply test/dshell/dll.d:8]: I presume  we do, cc @kinke?

Comment (kinke) [reply test/dshell/dll.d:8]: Yep, we do, and this test works with LDC on macOS, with a little adaptation (https://github.com/ldc-developers/dmd-testsuite/blob/04d6f174b71032ea2e7ba96db81ed136e6eb1329/dshell/dll.d#L27-L28).

**Review by thewilsonator — APPROVED** [review]

File: test/dshell/dll.d [hunk 2]
```diff
@@ -5,6 +5,13 @@ int main()
     version (Windows) if (Vars.MODEL == "32omf") // Avoid optlink
         return DISABLED;
 
+    version (DigitalMars)
+        version (OSX) // Shared libraries are not yet supported on OSX
+            return DISABLED;
+
+    version (OSX)
+        Vars.set(`SOEXT`, `.dylib`);
```
[the remark this answers is not in the document]
Comment (ibuclaw) [reply test/dshell/dll.d:13]: Alright.

**Review by kinke — APPROVED** [review]

**Pull request merged**